File: samples/src/commonMain/kotlin/sample/_common/BasicTypeMapping.kt
package sample._common

import deezer.kustomexport.KustomExport

@KustomExport
class BasicTypeMapping {
    val bool: Boolean = true
    val byte: Byte = 0x42
    // TODO: val char: Char = 'c' // Not supported anymore ???
    val short: Short = 0x7F7F
    val int: Int = 0x7F4251
    val float: Float = 12345.678f
    val double: Double = 123456789.123456789
    val string: String = "magic"
    val booleanArray: BooleanArray = BooleanArray(5) { it % 2 == 0 }
    val byteArray: ByteArray = ByteArray(5) { it.toByte() }
    val shortArray: ShortArray = ShortArray(5) { it.toShort() }
    val intArray: IntArray = IntArray(5) { it }
    val floatArray: FloatArray = FloatArray(5) { it.toFloat() }
    val doubleArray: DoubleArray = DoubleArray(5) { it.toDouble() }
    val charArray: CharArray = CharArray(5) { it.toChar() }

    // ...
    val any: Any = Any()
    // TODO: val unit: Unit = Unit // Not supported anymore ???
}

File: samples/src/commonMain/kotlin/sample/_common/BasicTypeMapping.ts
import { runTest } from "../shared_ts/RunTest"
import { assert } from "../shared_ts/Assert"
import { sample } from '@kustom/Samples'

runTest("BasicTypeMapping", () : void => {
    var basicTypes = new sample._common.js.BasicTypeMapping()
    assert(basicTypes.bool == true, "mapping for Boolean")
    assert(basicTypes.byte == 0x42, "mapping for Byte")
    // TODO : assert(basicTypes.char == 'c', "mapping for Char")
    assert(basicTypes.short == 0x7F7F, "mapping for Short")
    assert(basicTypes.int == 0x7F4251, "mapping for Int")
    assert(basicTypes.float == 12345.678, "mapping for Float")
    assert(basicTypes.double == 123456789.123456789, "mapping for Double")
    assert(basicTypes.string == "magic", "mapping for String")

    // Current export for typed array looks a bit off, need to check with typescript dev
    console.log("//TODO: Typed arrays")

    assert(JSON.stringify(basicTypes.any) == JSON.stringify(new Object()), "mapping for Any")
    // TODO : assert(JSON.stringify(basicTypes.unit) == "{}", "mapping for Unit")
})